### DISCUSSION TITLE
Issue with Routing and custom widget after upgrading dojo and node versions

### DISCUSSION THREAD

**Discussion by jithendraTR** [community/community#44671] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Bug

### Body

Hi Team,

We are facing couple of issue with dojo framework after we upgrade our node and npm version, please find the details below and it will be of great help if you could please provide inputs on how this can be resolved

Application structure: We are using dojo framework application build on node and npm, we are using grunt along with dojo and npm node modules
Issue: We started facing issues with grunt 2 months back without any changes to our code, we were not able to do npm install, logs were provided below for your reference

issue logs:
--------------------------------------------------------------------------------------
module.js:478
    throw err;
    ^
Error: Cannot find module 'has'
    at Function.Module._resolveFilename (module.js:476:15)
    at Function.Module._load (module.js:424:25)
    at Module.require (module.js:504:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (...\v6.14.4\node_modules\grunt-cli\node_modules\is-core-module\index.js:3:11)
    at Module._compile (module.js:577:32)
    at Object.Module._extensions..js (module.js:586:10)
    at Module.load (module.js:494:32)
    at tryModuleLoad (module.js:453:12)
    at Function.Module._load (module.js:445:3)
--------------------------------------------------------------------------------------
module.js:478
    throw err;
    ^
Error: Cannot find module 'ini'
    at Function.Module._resolveFilename (module.js:476:15)
    at Function.Module._load (module.js:424:25)
    at Module.require (module.js:504:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/jenkins/workspace/module/node_modules/global-prefix/index.js:14:11)
    at Module._compile (module.js:577:32)
    at Object.Module._extensions..js (module.js:586:10)
    at Module.load (module.js:494:32)
    at tryModuleLoad (module.js:453:12)
   at Function.Module._load (module.js:445:3)
--------------------------------------------------------------------------------------

Version we were using:
node: 6.14.4
npm: 3.10.10
grunt-cli: 0.1.13
python: 2.7
Visual studio: 2013

dojo node modules we were using:
"dijit": "dojo/dijit#1.9.9"
"dojo": "dojo/dojo#1.9.9"
"dojo-util": "dojo/util#1.10.4"
"dojox": "dojo/dojox#1.9.9"

Based on the suggestions received we have upgraded node and npm version along with grunt, dojo and some of the other node modules, please find the version we upgraded to
node: 16.13.2
npm: 8.19.3
grunt-cli: 1.4.3
python: 3.10
Visual studio: 2022

dojo node modules we updated:
"dijit": "^1.17.3"
"dojo": "^1.17.3"
"dojo-util": "^1.17.3"
"dojox": "^1.17.3"

After the above upgrade changes we are seeing couple of issues with routing and custom widgets
Issue 1:
We are using router.go function to navigate between pages, here we implemented replacements and query as parameter but from one of our view the navigation is not working, though the url path is getting changed on click of button the page is not refreshed and still shows the existing one, it stays on same page if same button is clicked but when other button is clicked the url path changes and navigation happens this time, issue is only with first time navigation and the same router.go implementation is working in other places in application.

We have tried resetting the router, reinitializing and reload, but reloading is refreshing whole page which is not a considerable solution.

Issue 2:
We have a multiselect calendar to select date with right and left arrows using custom widget dijit base css, we have writter our own implementation with table, tr and td, after the upgrade the table classes are not getting applied and calendar format is broken

Any inputs of suggestions here are much appreciated or if you can please provide what was changed between the versions we upgraded (might be some documentation if we can find some where, we tried looking at dojo community but couldn't find any) please reply if you need further information, Thanks a lot in advance.

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @jithendraTR ! 

We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Issues category is solely related to conversations around the GitHub products Issues and Projects. I've gone ahead and moved it for you. Good luck!
